api/pkg/apis/v1alpha1/providers/states/k8s/k8s.go

```diff
@@ -260,8 +262,18 @@ func (s *K8sStateProvider) Upsert(ctx context.Context, entry states.UpsertReques
 				sLog.Errorf("  P (K8s State): failed to update object: %v", err)
 				return "", err
 			}
+			getResourceVersion = true
 		}
 		if v, ok := dict["status"]; ok {
+			if getResourceVersion {
+				// Get latest resource version in case the the object spec is also updated
```

Comment:
Why we decide to get object again instead of getting error when upsert? If one object is updated and vendor/manager is not aware of the changes, behavior may be different.

Reply:
the manager may want to update spec and status in a call.

Reply:
This `get` may override `item.Object["spec"]`.

Reply:
This `get` may override `item.Object["spec"]`.